### ISSUE TITLE
Add support to disable premultipliedAlpha

### ISSUE THREAD

**Issue by w0rm** [elm-community/elm-webgl#19]
It is not possible to disable premultipliedAlpha which is very useful to have transparent sprites. 

Currently the transparency issue can only be solved by [hacking the shader code](https://github.com/zalando/elm-street-404/commit/ce7c84530c7f194c42518773f79716571f03c48e).

**Comment by Zinggi** [comment]
This is quite a bummer.
I'm currently working on [this little 2d rendering engine](https://github.com/Zinggi/elm-2d-game) and this would be very much needed for any 2d game with sprites.

I've read the contributing.md, but this can't be added as a PATCH.

Best would be to add something like

``` elm
toHtmlWithEvenMore
    :  { alpha: Bool, depth: Bool, stencil: Bool, antialias: Bool, premultipliedAlpha: Bool  }
    -> List FunctionCall
    -> List (Attribute msg)
    -> List Renderable
    -> Html msg
```

 (Info to these properties [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext))

This would be a MINOR change.

I left out two properties because I don't really understand the first one and because I think the second one would be useless in elm:
- preserveDrawingBuffer: If the value is true the buffers will not be cleared and will preserve their values until cleared or overwritten by the author.
- failIfMajorPerformanceCaveat: Boolean that indicates if a context will be created if the system performance is low.

I'm currently also doing the little shader trick from @w0rm combined with `BlendFunc (One, OneMinusSrcAlpha)`, but this isn't a good solution.
The problem with this solution is that you also have to change the rendering order. This makes rendering multiple objects very unintuitive for a user.

**Comment by nphollon** [comment]
Possible alternative to @Zinggi API suggestion: What if we extended the [Capability](http://package.elm-lang.org/packages/elm-community/elm-webgl/3.0.3/WebGL#Capability) type to cover these attributes?

Then there would be no need to add a new `toHtml`-type function... you could just call `Disable PremultipliedAlpha` to your `FunctionCall` list. This would be a less invasive API change.

I realize that premultipliedAlpha is not _actually_ a function call, but from an implementation perspective I don't see a problem with treating it like one. What would y'all think? Would this be confusing?

**Comment by Zinggi** [comment]
@nphollon, I like the suggestion, but there is a problem with it.

You can use `List FunctionCall` as an agrument to `renderWithConfig` too.
Now with your suggestion you could then write a nonsense `renderWithConfig [Enable PremultipliedAlpha]` which doesn't make sense on a per render basis, as it (premultipliedAlpha) only applies to the top-level canvas element.

It would be a way to avoid bumping the version, but it wouldn't be the best API.

I think someone has to bother Evan for this. It's not critical, but it would be a big improvement. I would go ahead and implement it, but first I have to know whether a pull request would even be accepted.

**Comment by nphollon** [comment]
Good point, I had forgotten about `renderWithConfig`. (It would have required a version bump anyway, since it would involve changing a type definition.)

As for whether the PR would be accepted, you can get approval from either @evancz or @johnpmayer. We also have #32 waiting for approval at the moment. I think it would make sense to release the updates together if they both get the thumbs up.

**Comment by Zinggi** [comment]
All right, I'll try to get my hands dirty on this one.
Will take some time though, as I haven't done native stuff yet.

**Comment by Zinggi** [comment]
Ok, so I implemented this here: https://github.com/elm-community/elm-webgl/pull/38

**Comment by w0rm** [comment]
This is implemented as the part of 2.0.
